libtenzir/builtins/formats/kv.cpp

```diff
@@ -124,43 +223,45 @@ class kv_parser final : public plugin_parser {
   auto
   instantiate(generator<chunk_ptr> input, operator_control_plane& ctrl) const
     -> std::optional<generator<table_slice>> override {
-    (void)input;
-    diagnostic::error("`{}` cannot be used here", name())
-      .emit(ctrl.diagnostics());
-    return {};
+    return parse_loop(to_lines(std::move(input)), ctrl, *this);
+  }
+
+  auto parse_line(series_builder& builder, operator_control_plane& ctrl,
+                  std::string_view line) const -> void {
+    auto event = builder.record();
+    while (not line.empty()) {
+      // TODO: We ignore split failures here. There might be better ways
+      // to handle this.
+      auto [head, tail] = field_split_.split(line);
+      auto [key_view, value_view] = value_split_.split(head);
+      auto key = unescape(trim_quotes(key_view));
+      auto value = unescape(trim_quotes(value_view));
```

Comment:
Do we want to unescape values that are not quoted? 

I am also thinking about things like `path=C:\something` and `path=C:\nothing`. I know that `\n` is not unescaped in your current implementation, and I just made a comment that says that we maybe should unescape it. Maybe the difference is that quotes indicate that a value should be unescaped? Unless `path="C:\nothing"` is something that tools would emit? Thoughts?

Reply:
A `C:\path` was indeed the reason why I did it manually.

I do think that simply limiting unescaping to quoted strings is the better choice though. We didnt do any unescaping previously, so that remains consistent.

I would hope that tools which emit quoted strings would also write properly escaped paths.

We could also add a flag to disable unescaping, or just add that flag if this ever becomes an issue.